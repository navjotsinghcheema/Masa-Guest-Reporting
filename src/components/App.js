import React from 'react';

import '../sass/index.scss';
import Container from 'react-bootstrap/Container'
import Navbar from 'react-bootstrap/Navbar'

let headerStyles = {
    margin: '0 auto',
    textTransform: 'uppercase',
    letterSpacing: '0.71rem'
};

const Header = () => (    
    <Navbar expand="lg" variant="light" bg="light" fixed="true">
        <Container>
            <Navbar.Brand href="#" style={headerStyles}>
                <span>Cafe</span> 
                <span style={{marginLeft: '10%'}}>Masa</span>
            </Navbar.Brand>
        </Container>
    </Navbar>
);

class App extends React.Component {  

    constructor(props) {
        super(props);        
    }        

    render() {
        return (
        <div id="App">   
            <Header />
            <div id="App-body" className="mt-5">
                <Container>
                    Application...
                </Container>                
            </div>            
        </div>
        );
    }
}

export default App;