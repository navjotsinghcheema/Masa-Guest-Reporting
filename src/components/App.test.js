import React from 'react';
import renderer from 'react-test-renderer';
import Guests from '../data/guests';
import App from './App';

test('App snapshot test', () => {
  const component = renderer.create(<App />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

describe("guests.json sanity tests", () => {
    test("Row count matches # of elements in data array", () => {           
      const numGuests = Guests["meta-data"]["row_count"];
      expect(Guests.data.length).toBe(numGuests);
    });
  
    test("Number of elements in data array are greater than 0", () => {
      expect(Guests.data.length).toBeGreaterThan(0);
    });
  
    // To ensure the data recieved is of the same format as meta-data payloadx
    test("Test data types of elements with first element", () => {
      const firstElementKeys = Object.keys(Guests.data[0]);
      const metadataElementKeys = Guests["meta-data"]["payload"].map(item => item.id);
      expect(firstElementKeys).toContain(...metadataElementKeys)
    })

    test("Data must contain a value for total spend and visit count as a positive number for each user", () => {
      Guests.data.map(guest => {
        expect(guest.visit_count).toBeGreaterThanOrEqual(0);
        expect(guest.total_spend).toBeGreaterThanOrEqual(0);
      })
    })
  })