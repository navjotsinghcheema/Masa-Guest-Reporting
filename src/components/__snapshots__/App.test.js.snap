// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot test 1`] = `
<div
  id="App"
>
  <nav
    className="navbar navbar-expand-lg navbar-light bg-light fixed-true"
  >
    <div
      className="container"
    >
      <a
        className="navbar-brand"
        href="#"
        style={
          Object {
            "letterSpacing": "0.71rem",
            "margin": "0 auto",
            "textTransform": "uppercase",
          }
        }
      >
        <span>
          Cafe
        </span>
        <span
          style={
            Object {
              "marginLeft": "10%",
            }
          }
        >
          Masa
        </span>
      </a>
    </div>
  </nav>
  <div
    className="mt-5"
    id="App-body"
  >
    <div
      className="container"
    >
      Application...
      <div
        className="row"
      >
        <div
          className="col-lg-4 col-md-4 col-sm-6 col-xs-12"
        >
          <div
            className="col-lg-12 col-md-12 col-sm-12 col-xs-12"
          >
            <div
              className="card"
            >
              <div
                className="card-body"
              >
                <h5
                  className="card-title"
                >
                  <strong>
                    Firstname
                  </strong>
                   Lastname
                </h5>
              </div>
            </div>
            <div
              className="card"
            >
              <div
                className="card-body"
              >
                <h5
                  className="card-title"
                >
                  <strong>
                    Firstname
                  </strong>
                   Lastname
                </h5>
              </div>
            </div>
            <div
              className="card"
            >
              <div
                className="card-body"
              >
                <h5
                  className="card-title"
                >
                  <strong>
                    Firstname
                  </strong>
                   Lastname
                </h5>
              </div>
            </div>
            <div
              className="card"
            >
              <div
                className="card-body"
              >
                <h5
                  className="card-title"
                >
                  <strong>
                    Firstname
                  </strong>
                   Lastname
                </h5>
              </div>
            </div>
            <div
              className="card"
            >
              <div
                className="card-body"
              >
                <h5
                  className="card-title"
                >
                  <strong>
                    Firstname
                  </strong>
                   Lastname
                </h5>
              </div>
            </div>
            <div
              className="card"
            >
              <div
                className="card-body"
              >
                <h5
                  className="card-title"
                >
                  <strong>
                    Firstname
                  </strong>
                   Lastname
                </h5>
              </div>
            </div>
            <div
              className="card"
            >
              <div
                className="card-body"
              >
                <h5
                  className="card-title"
                >
                  <strong>
                    Firstname
                  </strong>
                   Lastname
                </h5>
              </div>
            </div>
            <div
              className="card"
            >
              <div
                className="card-body"
              >
                <h5
                  className="card-title"
                >
                  <strong>
                    Firstname
                  </strong>
                   Lastname
                </h5>
              </div>
            </div>
            <div
              className="card"
            >
              <div
                className="card-body"
              >
                <h5
                  className="card-title"
                >
                  <strong>
                    Firstname
                  </strong>
                   Lastname
                </h5>
              </div>
            </div>
            <div
              className="card"
            >
              <div
                className="card-body"
              >
                <h5
                  className="card-title"
                >
                  <strong>
                    Firstname
                  </strong>
                   Lastname
                </h5>
              </div>
            </div>
            <div
              className="card"
            >
              <div
                className="card-body"
              >
                <h5
                  className="card-title"
                >
                  <strong>
                    Firstname
                  </strong>
                   Lastname
                </h5>
              </div>
            </div>
            <div
              className="card"
            >
              <div
                className="card-body"
              >
                <h5
                  className="card-title"
                >
                  <strong>
                    Firstname
                  </strong>
                   Lastname
                </h5>
              </div>
            </div>
            <div
              className="card"
            >
              <div
                className="card-body"
              >
                <h5
                  className="card-title"
                >
                  <strong>
                    Firstname
                  </strong>
                   Lastname
                </h5>
              </div>
            </div>
            <div
              className="card"
            >
              <div
                className="card-body"
              >
                <h5
                  className="card-title"
                >
                  <strong>
                    Firstname
                  </strong>
                   Lastname
                </h5>
              </div>
            </div>
            <div
              className="card"
            >
              <div
                className="card-body"
              >
                <h5
                  className="card-title"
                >
                  <strong>
                    Firstname
                  </strong>
                   Lastname
                </h5>
              </div>
            </div>
            <div
              className="card"
            >
              <div
                className="card-body"
              >
                <h5
                  className="card-title"
                >
                  <strong>
                    Firstname
                  </strong>
                   Lastname
                </h5>
              </div>
            </div>
            <div
              className="card"
            >
              <div
                className="card-body"
              >
                <h5
                  className="card-title"
                >
                  <strong>
                    Firstname
                  </strong>
                   Lastname
                </h5>
              </div>
            </div>
            <div
              className="card"
            >
              <div
                className="card-body"
              >
                <h5
                  className="card-title"
                >
                  <strong>
                    Firstname
                  </strong>
                   Lastname
                </h5>
              </div>
            </div>
            <div
              className="card"
            >
              <div
                className="card-body"
              >
                <h5
                  className="card-title"
                >
                  <strong>
                    Firstname
                  </strong>
                   Lastname
                </h5>
              </div>
            </div>
            <div
              className="card"
            >
              <div
                className="card-body"
              >
                <h5
                  className="card-title"
                >
                  <strong>
                    Firstname
                  </strong>
                   Lastname
                </h5>
              </div>
            </div>
            <div
              className="card"
            >
              <div
                className="card-body"
              >
                <h5
                  className="card-title"
                >
                  <strong>
                    Firstname
                  </strong>
                   Lastname
                </h5>
              </div>
            </div>
            <div
              className="card"
            >
              <div
                className="card-body"
              >
                <h5
                  className="card-title"
                >
                  <strong>
                    Firstname
                  </strong>
                   Lastname
                </h5>
              </div>
            </div>
            <div
              className="card"
            >
              <div
                className="card-body"
              >
                <h5
                  className="card-title"
                >
                  <strong>
                    Firstname
                  </strong>
                   Lastname
                </h5>
              </div>
            </div>
            <div
              className="card"
            >
              <div
                className="card-body"
              >
                <h5
                  className="card-title"
                >
                  <strong>
                    Firstname
                  </strong>
                   Lastname
                </h5>
              </div>
            </div>
            <div
              className="card"
            >
              <div
                className="card-body"
              >
                <h5
                  className="card-title"
                >
                  <strong>
                    Firstname
                  </strong>
                   Lastname
                </h5>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
